File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/production/unit-measure/unit-measure-lookup-model.ts

export interface IUnitMeasureLookupModel {
  unitMeasureCode: string;
  name: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class UnitMeasureLookupModel implements IUnitMeasureLookupModel {
  unitMeasureCode: string;
  name: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<UnitMeasureLookupModel>) {
    if (!!init) {
      Object.assign<UnitMeasureLookupModel, Partial<UnitMeasureLookupModel>>(this, init);
    }
  }
}

File: src/AdventureWorks.Application/DataEngine/Production/UnitMeasure/Commands/Queries/GetUnitMeasures/UnitMeasureLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Production;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;
using AdventureWorks.Application.DataEngine.Purchasing.ProductVendor.Queries.GetProductVendors;

namespace AdventureWorks.Application.DataEngine.Production.UnitMeasure.Queries.GetUnitMeasures
{
    public partial class UnitMeasureLookupModel : IHaveCustomMapping
    {
        public string UnitMeasureCode { get; set; }
        public string Name { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Production.UnitMeasure, UnitMeasureLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<UnitMeasureLookupModel, BaseUnitMeasure>().IgnoreMissingDestinationMembers();
        }
    }
}
